### PR TITLE
Ref actions error messages

### LINKED ISSUE #1974: Referential actions - Changes in error messages when deletes are prevented to preserve referential integrity
When Prisma prevents deletion of records to preserve referential integrity, e.g. using `delete()` or `deleteMany()` on an object with child records on a required relation, it throws a custom runtime error.

When opting-in to the referential actions features (2.26.0 and higher), Prisma no longer performs any runtime checks. Developers can instead specify custom referential actions on the relations in their schema and Prisma will leverage the foreign key constraints to preserve referential integrity. When using `NoAction` or `Restrict` to prevent deletion, the errors will be different as they are triggered by the database and not by Prisma Client.

This should be documented as a note and we should include a section in our upgrade docs so that developers make sure to adjust their code to look for other errors if they are using these for control flow.

Closes https://github.com/prisma/prisma/issues/8054

### PR DESCRIPTION
**Pull request by molebox**
This PR adds a new section under the upgrade paths about error handling pre 2.26.0 and post 2.26.0

Fixes: #1974

### CODE REVIEW

File: content/200-concepts/100-components/01-prisma-schema/06-relations/07-referential-actions.mdx [hunk 1]
```diff
@@ -405,6 +405,105 @@ Explicitly defining referential actions in your Prisma schema is optional. If yo
...
+  try {
+    await prisma.user.delete({
+      where: {
+        id: 'some-long-id'
+      }
+    })
+  } catch (error) {
+    if (error instanceof Prisma.PrismaClientKnownRequestError) {
+      if (error.code === 'P2014') {
+      if (error.code === 'P2003') {
```
Comment (Jolg42) [inline content/200-concepts/100-components/01-prisma-schema/06-relations/07-referential-actions.mdx:491]: This is not valid code, maybe you wanted to remove one line here?
```suggestion
      if (error.code === 'P2003') {
```

File: content/200-concepts/100-components/01-prisma-schema/06-relations/07-referential-actions.mdx [hunk 2]
```diff
@@ -405,6 +405,105 @@ Explicitly defining referential actions in your Prisma schema is optional. If yo
 
 Note that referential actions can be added on a case by case basis. This means that you can add them to one single relation and leave the rest set to the defaults by not manually specifying anything.
 
+### Checking for errors
+
+**Before** upgrading to 2.26.0 and enabling the referential actions **preview feature**, Prisma prevented the deletion of records while using `delete()` or `deleteMany()` to preserve referential integrity. A custom runtime error would be thrown by the Prisma Client.
```
Comment (janpio) [inline content/200-concepts/100-components/01-prisma-schema/06-relations/07-referential-actions.mdx]: I think we could mention the error code here, as it is a specific one.

File: content/200-concepts/100-components/01-prisma-schema/06-relations/07-referential-actions.mdx [hunk 3]
```diff
@@ -405,6 +405,105 @@ Explicitly defining referential actions in your Prisma schema is optional. If yo
...
+
+**After** upgrading and enabling the referential actions **preview feature**, Prisma no longer performs runtime checks. You can instead specify a custom referential action to preserve the referential integrity between relations. 
+
+When you use [`NoAction`](#noaction) or [`Restrict`](#restrict) to prevent the deletion of records, the error messages will be different post 2.26.0 compared to pre 2.26.0. This is because they are now triggered by the database and **not** the Prisma Client.
```
Comment (janpio) [inline content/200-concepts/100-components/01-prisma-schema/06-relations/07-referential-actions.mdx]: I think we can also mention the common eror here.

File: content/200-concepts/100-components/01-prisma-schema/06-relations/07-referential-actions.mdx (hunk 1 above)
Comment (molebox) [reply content/200-concepts/100-components/01-prisma-schema/06-relations/07-referential-actions.mdx:491]: Looks like i mis-counted the lines, its supposed to delete one line and add another. Good ctach

Comment (janpio) [inline content/200-concepts/100-components/01-prisma-schema/06-relations/07-referential-actions.mdx:491]: Totally unrelated to this PR: 
This is why I prefer the old way to highlight this diff with `+` and `-` - on first sight this looks like invalid, broken code and made me do a double take until I realized the `highlight` above.

**Review by janpio — APPROVED** [review]
👍

File: content/200-concepts/100-components/01-prisma-schema/06-relations/07-referential-actions.mdx (hunk 1 above)
Comment (molebox) [reply content/200-concepts/100-components/01-prisma-schema/06-relations/07-referential-actions.mdx:491]: Yeah, i agree its not pretty. Ironically i mis-counted the lines too

Comment (janpio) [reply content/200-concepts/100-components/01-prisma-schema/06-relations/07-referential-actions.mdx:491]: Haha, for real? Nice :D

Comment (Jolg42) [reply content/200-concepts/100-components/01-prisma-schema/06-relations/07-referential-actions.mdx:491]: Oh this highlight at the top is 😕, not used to it I guess. The +- are indeed easier to read.

**Pull request merged**